src/components/Card.js

```diff
@@ -1,10 +1,42 @@
 export class Card {
-  constructor(name, link, cardSelector, handleImageClick) {
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    handleImageClick,
+    handleDelete,
+    handleLikeToggle
+  ) {
     this._name = name;
     this._link = link;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDelete = handleDelete;
+    this._handleLikeToggle = handleLikeToggle;
+    this._id = _id;
+    this._isLiked = isLiked;
   }
+
+  getCardId() {
+    return this._id;
+  }
+
+  getLikedState() {
+    return this._isLiked;
+  }
+  updateLikes(newLikes) {
```

Comment:
The logic for updating and rendering likes looks brilliant!

Reply:
thank you so much